includes/class-wc-gateway-swedbank-pay-cc.php

```diff
@@ -813,331 +769,119 @@ public function return_handler() {
 
 			// Create Background Process Task
 			$background_process = new WC_Background_Swedbank_Pay_Queue();
-			$background_process->push_to_queue( [
-				'payment_method_id' => $this->id,
-				'webhook_data'      => $raw_body,
-			] );
+			$background_process->push_to_queue(
+				array(
+					'payment_method_id' => $this->id,
+					'webhook_data'      => $raw_body,
+				)
+			);
 			$background_process->save();
 
-			$this->log( sprintf( 'Incoming Callback: Task enqueued. Transaction ID: %s', $data['transaction']['number'] ) );
+			$this->adapter->log(
+				LogLevel::INFO,
+				sprintf( 'Incoming Callback: Task enqueued. Transaction ID: %s', $data['transaction']['number'] )
+			);
 		} catch ( Exception $e ) {
-			$this->log( sprintf( 'Incoming Callback: %s', $e->getMessage() ) );
+			$this->adapter->log( LogLevel::INFO, sprintf( 'Incoming Callback: %s', $e->getMessage() ) );
 		}
 	}
 
 	/**
-	 * Check is Capture possible
-	 *
-	 * @param WC_Order|int $order
-	 * @param bool $amount
+	 * Process Refund
 	 *
-	 * @return bool
-	 */
-	public function can_capture( $order, $amount = false ) {
-		if ( is_int( $order ) ) {
-			$order = wc_get_order( $order );
-		}
-
-		if ( ! $amount ) {
-			$amount = $order->get_total();
-		}
-
-		$state = $order->get_meta( '_payex_payment_state' );
-		if ( empty( $state ) || $state === 'Authorized' ) {
-			return true;
-		}
-
-		return false;
-	}
-
-	/**
-	 * Check is Cancel possible
+	 * If the gateway declares 'refunds' support, this will allow it to refund
+	 * a passed in amount.
 	 *
-	 * @param WC_Order|int $order
+	 * @param int $order_id
+	 * @param float $amount
+	 * @param string $reason
 	 *
-	 * @return bool
+	 * @return  bool|wp_error True or false based on success, or a WP_Error object
 	 */
-	public function can_cancel( $order ) {
-		if ( is_int( $order ) ) {
-			$order = wc_get_order( $order );
-		}
-
-		$state = $order->get_meta( '_payex_payment_state' );
-
-		if ( in_array( $state, [
-			'Captured',
-			'Cancelled',
-			'Refunded'
-		] ) ) {
+	public function process_refund( $order_id, $amount = null, $reason = '' ) {
+		$order = wc_get_order( $order_id );
+		if ( ! $order ) {
 			return false;
 		}
 
-		return true;
-	}
-
-	/**
-	 * @param \WC_Order $order
-	 * @param bool $amount
-	 *
-	 * @return bool
-	 * @throws \Exception
-	 */
-	public function can_refund( $order, $amount = false ) {
-		if ( is_int( $order ) ) {
-			$order = wc_get_order( $order );
-		}
-
-		if ( ! $amount ) {
+		// Full Refund
+		if ( is_null( $amount ) ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function is_null() is discouraged; use strict comparison "=== null" instead.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)